src/checkpoint/dispatch/dispatch.impl.h

```diff
@@ -122,7 +122,9 @@ T* Standard::construct(SerialByteType* mem) {
 
 template <typename Serializer, typename T>
 inline void serializeArray(Serializer& s, T* array, SerialSizeType const len) {
-  Traverse::with<T, Serializer>(*array, s, len);
+  if (array) {
```

Comment:
I'm not sure if this is enough:

> https://en.cppreference.com/w/cpp/container/vector/data
If size() is 0, data() may or may not return a null pointer.

Reply:
How about `if (len > 0)`?

Reply:
ad53ede crashed because with non-null pointer, but `if (len > 0)` seems to work